go-controller/pkg/ovn/master.go

```diff
@@ -1491,3 +1499,233 @@ func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
 	}
 	return nil
 }
+
+func (oc *Controller) addNodeObj(node *kapi.Node) error {
+	var gatewaysFailed sync.Map
+	var mgmtPortFailed sync.Map
+	var addNodeFailed sync.Map
+	var nodeClusterRouterPortFailed sync.Map
+	var errs []error
+
+	if noHostSubnet := noHostSubnet(node); noHostSubnet {
+		err := oc.lsManager.AddNoHostSubnetNode(node.Name)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	klog.Infof("Adding Node %q", node.Name)
+	hostSubnets, err := oc.addNode(node)
+	if err != nil {
+		klog.Errorf("NodeAdd: error creating subnet for node %s: %v", node.Name, err)
+		addNodeFailed.Store(node.Name, true)
+		nodeClusterRouterPortFailed.Store(node.Name, true)
+		mgmtPortFailed.Store(node.Name, true)
+		gatewaysFailed.Store(node.Name, true)
+		errs = append(errs, err)
+	}
```

Comment:
if `oc.addNode(node)` fails, should we really go on with the following calls (`oc.syncNodeClusterRouterPort`, `oc.syncNodeManagementPort`, etc.)? I see that the original code returned if `addNode` failed.

Reply:
u are right I should have left those early 2 return and just aggregate the other cases